python/pyspark/mllib/classification.py

```diff
@@ -237,12 +290,19 @@ def train(cls, data, iterations=100, initialWeights=None, regParam=0.01, regType
         def train(rdd, i):
             return callMLlibFunc("trainLogisticRegressionModelWithLBFGS", rdd, int(iterations), i,
                                  float(regParam), regType, bool(intercept), int(corrections),
-                                 float(tolerance))
-
+                                 float(tolerance), bool(validateData), int(numClasses))
+
+        if initialWeights is None:
+            if numClasses == 2:
+                initialWeights = [0.0] * len(data.first().features)
+            elif intercept:
```

Comment:
Would you mind putting the "if intercept" in an "else" clause to make it clearer which part of this is for multiclass?
```
if numClasses == 2:
else:
  if intercept:
  else:
```